src/app/plugins/platform/Docker/plugins/cms/WordPress/index.ts

```diff
@@ -328,8 +335,61 @@ class WordPress extends Generator {
     }
   }
 
+  private async _installGessoDependencies() {
+    if (!this.shouldInstall) {
+      return;
+    }
+
+    // Install required dependencies to avoid Gesso crashing when enabled
+    for (const dependency of gessoWPDependencies) {
+      // Use Composer if we can.
+      if (this.usesWpStarter) {
```

Comment:
I'd like to see the two branches of this conditional broken out into their own functions. We don't need distracting implementation details in the middle of the loop, and factoring them out means that we can simply pick one of the two installation methods before the loop event starts, rather than having to choose at each iteration.

I'd suggest something like this:

```ts
// Names subject to bikeshedding
const install: (dep: string) => Promise<void> = this.usesWpStarter
  ? dep => this._installViaComposer(dep)
  : dep => this._installViaZip(dep); 

for (const dependency of gessoWPDependencies) {
  await install(dep);
}
```

Reply:
Ah, yeah, good point about avoiding choosing the install method with each loop iteration. (It's not like it's going to change midway through...) I like your suggested approach -- I'll refactor later this week.

Reply:
Updated!